diff --git a/src/rimestate.cpp b/src/rimestate.cpp
index bfe3aec..94b8d28 100644
--- a/src/rimestate.cpp
+++ b/src/rimestate.cpp
@@ -113,12 +113,6 @@ std::string RimeState::subModeLabel() {
             result = asciiModeName(/*abbrev=*/true);
         } else if (status.schema_name && status.schema_name[0] != '.') {
             result = status.schema_name;
-            if (!result.empty() &&
-                utf8::lengthValidated(result) != utf8::INVALID_LENGTH) {
-                result = result.substr(
-                    0, std::distance(result.begin(),
-                                     utf8::nextChar(result.begin())));
-            }
         }
     });
     return result;
